wandb/old/settings.py

```diff
@@ -163,6 +164,11 @@ def get_username() -> str:
                     try_create_dir(config_dir)
                 else:
                     config_dir = temp_config_dir
+                wandb.termwarn(
+                    f"Error creating global config settings in: {home_config_dir}."
```

Comment:
```suggestion
                    f"Failed to create global config settings in: {home_config_dir}."
```

This warning should be logged at the start of the conditional. Otherwise you have to read through the nested conditionals in case anything returns or sets `home_config_dir` to trust the behavior.